sdks/python/apache_beam/io/localfilesystem.py

```diff
@@ -295,11 +296,20 @@ def _delete_path(path):
         raise IOError(err)
 
     exceptions = {}
-    for path in paths:
+
+    def try_delete(path):
       try:
         _delete_path(path)
       except Exception as e:  # pylint: disable=broad-except
         exceptions[path] = e
 
+    for path in paths:
+      expanded = glob(path)
+      if not expanded:
+        try_delete(path)
+      else:
+        for p in expanded:
```

Comment:
I believe we should be relying on `filesystem.match` for globbing to have a consistent *globbing* experience across the filesystem APIs. This would be similar to what we do within the [gcsfilesystem.py](https://github.com/apache/beam/blob/1348cae2daaa403867fba192e4e06ff4addbf5ae/sdks/python/apache_beam/io/gcp/gcsfilesystem.py#L313)

Reply:
good idea, done